src/string.rs

```diff
@@ -65,6 +81,15 @@ impl From<Rc<str>> for IString {
     }
 }
 
+impl From<Cow<'static, str>> for IString {
+    fn from(cow: Cow<'static, str>) -> Self {
+        match cow {
+            Cow::Borrowed(s) => IString::Static(s),
+            Cow::Owned(s) => s.into(),
+        }
+    }
+}
```

Comment:
I think you forgot to write a test for this. It's not super important but it makes sure that someone does not remove it by mistake